GetStream/stream-chat-android
# [CAS-1087] Fix filtering by array extra data field

https://stream-io.atlassian.net/browse/CAS-1087

### 🎯 Goal

`in` and `nin` filters should work with extra data fields that are an array.

`Filters.`in`("extra_user_id_list", listOf(userId))`

### 🛠 Implementation details

Handling the case when extra data field is an array 

### 🧪 Testing

1. Create a named channel with  an array extra data field:

```
val memberIds = members.map { it.id } + currentUserId
ChatClient.instance()
                .createChannel(
                    channelType = CHANNEL_TYPE_MESSAGING,
                    channelId = UUID.randomUUID().toString(),
                    members = memberIds,
                    extraData = mapOf(EXTRA_DATA_CHANNEL_NAME to name, "extra_user_id_list" to memberIds)
                ).await()
```

2. Use the following filter:

```
private val filter: FilterObject = Filters.and(
        Filters.eq("type", "messaging"),
        Filters.`in`("extra_user_id_list", listOf(currentUserId)),
        Filters.`in`("members", listOf(currentUserId)),
        Filters.eq("frozen", false),
),
```

3. Observe that newly created channel is on the channel list

### ☑️ Checklist


### 🎉 GIF

![giphy (1)](https://user-images.githubusercontent.com/9600921/122500420-6f8a2f00-cffb-11eb-8f44-96b4e9be64a2.gif)
